celestial.js

```diff
@@ -1,1229 +1,1259 @@
-        if (clip(d.geometry.coordinates) && d.properties.mag <= cfg.stars.limit) {
-          var pt = mapProjection(d.geometry.coordinates),
-              r = starSize(d);
-          context.fillStyle = starColor(d); 
-          context.beginPath();
-          context.arc(pt[0], pt[1], r, 0, 2 * Math.PI);
-          context.closePath();
-          context.fill();
-          if (cfg.stars.designation && d.properties.mag <= cfg.stars.designationLimit*adapt) {
-            setTextStyle(cfg.stars.designationStyle);
-            context.fillText(starDesignation(d.id), pt[0]+r, pt[1]);
+      if (cfg.constellations.bounds) {
+        container.selectAll(parentElement + " .boundaryline").each(function (d) {
+          setStyle(cfg.constellations.boundStyle);
+          if (Celestial.constellation) {
+            var re = new RegExp("\\b" + Celestial.constellation + "\\b");
+            if (d.ids.search(re) !== -1) {
+              context.lineWidth *= 1.5;
+              context.globalAlpha = 1;
+              context.setLineDash([]);
+            }
+          }
+          map(d);
+          context.stroke();
+        });
+        context.setLineDash([]);
+      }
+
+      if (cfg.constellations.lines) {
+        container.selectAll(parentElement + " .constline").each(function (d) {
+          setStyleA(d.properties.rank, cfg.constellations.lineStyle);
+          map(d);
+          context.stroke();
+        });
+      }
+
+      drawOutline(true);
+
+      if (cfg.constellations.names) {
+        //setTextStyle(cfg.constellations.nameStyle);
+        container.selectAll(parentElement + " .constname").each(function (d) {
+          if (clip(d.geometry.coordinates)) {
+            setStyleA(d.properties.rank, cfg.constellations.nameStyle);
+            var pt = mapProjection(d.geometry.coordinates);
+            context.fillText(constName(d), pt[0], pt[1]);
           }
-          if (cfg.stars.propername && d.properties.mag <= cfg.stars.propernameLimit*adapt) {
-            setTextStyle(cfg.stars.propernameStyle);
-            context.fillText(starPropername(d.id), pt[0]-r, pt[1]);
+        });
+      }
+
+      if (cfg.stars.show) {
+        setStyle(cfg.stars.style);
+        container.selectAll(parentElement + " .star").each(function (d) {
+          if (clip(d.geometry.coordinates) && d.properties.mag <= cfg.stars.limit) {
+            var pt = mapProjection(d.geometry.coordinates),
+              r = starSize(d);
+            context.fillStyle = starColor(d);
+            context.beginPath();
+            context.arc(pt[0], pt[1], r, 0, 2 * Math.PI);
+            context.closePath();
+            context.fill();
+
+            // Draw the star
+            context.beginPath();
+            context.arc(pt[0], pt[1], r, 0, 2 * Math.PI);
+            context.closePath();
+            context.fill();
+
+            // Add glow effect
+            var starGlowColor = starColor(d); // get the color of the star
```

Comment:
zoom yapınca adapt değişkkeni 1 ile ~3 arasında değişiyor. 5 aşamalı draw yaparak glow yapınca arada çizgiler görünüyor, 15 yapınca da biraz kasıyor onun için zoom ile değişen bir şey yaptım.